File: python/semantic_kernel/prompt_template/input_variable.py
# Copyright (c) Microsoft. All rights reserved.

from typing import Any

from semantic_kernel.kernel_pydantic import KernelBaseModel


class InputVariable(KernelBaseModel):
    """Input variable for a prompt template.

    Args:
        name: The name of the input variable.
        description: The description of the input variable.
        default: The default value of the input variable.
        is_required: Whether the input variable is required.
        json_schema: The JSON schema for the input variable.
        allow_dangerously_set_content: Allow content without encoding, this controls
            if this variable is encoded before use, default is False.
    """

    name: str
    description: str | None = ""
    default: Any | None = ""
    is_required: bool | None = True
    json_schema: str | None = ""
    allow_dangerously_set_content: bool = False

File: nodejs/semantic-kernel/prompt-template/input-variable.ts
/**
 * Configuration for an input variable in a prompt template.
 */
export interface InputVariableConfig {
  /**
   * The name of the input variable.
   */
  name: string

  /**
   * The description of the input variable.
   */
  description?: string

  /**
   * The default value of the input variable.
   */
  default?: any

  /**
   * Whether the input variable is required.
   */
  isRequired?: boolean

  /**
   * The JSON schema for the input variable.
   */
  jsonSchema?: string

  /**
   * Allow content without encoding. This controls if this variable is encoded before use.
   * Default is false.
   */
  allowDangerouslySetContent?: boolean
}

/**
 * Input variable for a prompt template.
 */
export class InputVariable {
  /**
   * The name of the input variable.
   */
  public name: string

  /**
   * The description of the input variable.
   */
  public description: string

  /**
   * The default value of the input variable.
   */
  public default: any

  /**
   * Whether the input variable is required.
   */
  public isRequired: boolean

  /**
   * The JSON schema for the input variable.
   */
  public jsonSchema: string

  /**
   * Allow content without encoding. This controls if this variable is encoded before use.
   */
  public allowDangerouslySetContent: boolean

  /**
   * Creates a new InputVariable instance.
   * @param config - The configuration for the input variable.
   */
  constructor(config: InputVariableConfig) {
    this.name = config.name
    this.description = config.description ?? ''
    this.default = config.default ?? ''
    this.isRequired = config.isRequired ?? true
    this.jsonSchema = config.jsonSchema ?? ''
    this.allowDangerouslySetContent = config.allowDangerouslySetContent ?? false
  }
}
